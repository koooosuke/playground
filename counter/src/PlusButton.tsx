import React, { FC, useCallback } from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#008080',
    width: 50,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttnText: {
    color: 'white',
    fontSize: 20,
  },
});

interface Props {
  count: number;
  setCounter: (count: number) => void;
}

const PlusButton: FC<Props> = ({ count, setCounter }) => {
  const addCounter = useCallback(() => {
    setCounter(count + 1);
  }, [count, setCounter]);
  return (
    <TouchableOpacity style={styles.button} onPress={addCounter}>
      <Text style={styles.buttnText}>+</Text>
    </TouchableOpacity>
  );
};

export default PlusButton;
