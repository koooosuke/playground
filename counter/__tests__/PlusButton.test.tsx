import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import PlusButton from '../src/PlusButton';

describe('PlusButton', () => {
  const count = 10;
  it('通常のPlusButtonを描画できる', () => {
    const component = renderer.create(<PlusButton count={count} setCounter={() => {}} />);
    expect(component).toMatchSnapshot();
  });
  it(`ボタンを押下でカウントが${count}から${count + 1}になる`, () => {
    const onPressEvent = jest.fn();
    const { getByText } = render(<PlusButton count={count} setCounter={onPressEvent} />);
    fireEvent.press(getByText('+'));
    expect(onPressEvent).toBeCalled();
    expect(onPressEvent.mock.calls[0][0]).toBe(count + 1);
  });
});
