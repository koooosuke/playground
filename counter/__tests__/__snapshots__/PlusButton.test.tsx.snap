// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusButton 通常のPlusButtonを描画できる 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#008080",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    +
  </Text>
</View>
`;
